src/vm.rs

```diff
@@ -545,9 +545,8 @@ pub trait Vm {
 		}
 
 		// Verify that this module is a HermitCore ELF executable.
-		if elf.header.e_type != ET_DYN {
-			return Err(Error::InvalidFile(self.kernel_path().into()));
-		}
+
+		let is_dyn = elf.header.e_type == ET_DYN;
```

Comment:
Is that comment still valid? I don't see any code that actually verifies something? The check after your new line actually only checks the instruction set of the elf? Wasn't there a way to actually verify if it is a valid HC executable?

Reply:
I agree. I will remove the comment.